tools/src/keygen/KeygenOptions.hs

```diff
@@ -29,15 +31,18 @@ data KeygenOptions = KeygenOptions
     , koConfigurationOptions :: ConfigurationOptions
     } deriving (Show)
 
+newtype DgdPath      = DgdPath FilePath deriving (Show)
+
+newtype DgdCanonical = DgdCanonical Bool deriving (Show)
```

Comment:
For `newtype` over `Bool`, I typically like to just make a one-off type like

```
data DgdCanonical = DgdNonCanonical | DgdCanonical
```

Though, you can get similarly readable with `newtype DbgCanonical = DgdIsCanonical Bool`, as `DgcIsCanonical True` seems fairly clear to me

Reply:
Thanks for pointing this out.